streams/dshr_methods_mod.F90

```diff
@@ -44,17 +44,24 @@ subroutine dshr_state_getfldptr(State, fldname, fldptr1, fldptr2, rc)
     integer          ,          intent(out)             :: rc
 
     ! local variables
-    type(ESMF_Field)           :: lfield
+    type(ESMF_Field)            :: lfield
+    integer                     :: itemCount 
     character(len=*), parameter :: subname='(dshr_state_getfldptr)'
     ! ----------------------------------------------
 
     rc = ESMF_SUCCESS
 
-    call ESMF_StateGet(State, itemName=trim(fldname), field=lfield, rc=rc)
+    call ESMF_StateGet(State, itemSearch=trim(fldname), itemCount=itemCount, rc=rc)
     if (chkerr(rc,__LINE__,u_FILE_u)) return
 
-    call dshr_field_getfldptr(lfield, fldptr1=fldptr1, fldptr2=fldptr2, rc=rc)
-    if (chkerr(rc,__LINE__,u_FILE_u)) return
+    ! check field is in the state or not?
+    if (itemCount >= 1) then
+      call ESMF_StateGet(State, itemName=trim(fldname), field=lfield, rc=rc)
+      if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+      call dshr_field_getfldptr(lfield, fldptr1=fldptr1, fldptr2=fldptr2, rc=rc)
+      if (chkerr(rc,__LINE__,u_FILE_u)) return
+    end if
```

Comment:
We should document that the field pointer is not found - and so the field is not associated. Not sure where this documentation should be. For now this is okay - but it should also be an issue to implement.

Reply:
@mvertens You mean adding ESMF Log in else statement.

Reply:
That would be a start. I was wondering if we should have an output to logunit - or at least a comment in the code that says that the call to fldbun_getfldptr just returns if it cannot find the field. At least that way the code will make more sense. Does that sound reasonable? I don't want to have this hold up the PR.

Reply:
I put extra log into the code. Now, it prints following into the PET logs

```
20200729 112607.507 INFO             PET00   (dshr_state_getfldptr) Could not find the field: Sa_wspd just returning
20200729 112607.508 INFO             PET00   (dshr_state_getfldptr) Could not find the field: Faxa_lwnet just returning
```